### PR TITLE
[Feature] Support alter table syntax for sequence column 

### LINKED ISSUE #4560: Support Alter Table syntax for SEQUENCE COLUMN
**Is your feature request related to a problem? Please describe.**
Now we already support create table and load data with sequence column (#4256), we hope support alter table syntax for this function

**Describe the solution you'd like**
The Alter Table syntax like this  
`alter table test_tbl  enable feature "sequence column" 
with properties ("function_column.sequence_type" = 'Date');`

**Additional context**
this function only support UNIQUE_KEYS Table

If you have any Suggestions on this, please let me know, thx

### PR DESCRIPTION
**Pull request by Youngwb**
## Proposed changes

Support alter table syntax for sequence column

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #4560 ), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/EnableFeatureClause.java [hunk 1]
```diff
@@ -21,25 +21,35 @@
...
 import org.apache.logging.log4j.Logger;
 
+import java.util.Map;
+
 public class EnableFeatureClause extends AlterTableClause {
     private static final Logger LOG = LogManager.getLogger(EnableFeatureClause.class);
 
     public enum Features {
         BATCH_DELETE,
+        SEQUENCE_COLUMN,
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/EnableFeatureClause.java:35]: I think SEQUENCE_LOAD is better

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -981,6 +981,10 @@ alter_table_clause ::=
     {:
         RESULT = new EnableFeatureClause(featureName);
     :}
+    | KW_ENABLE KW_FEATURE STRING_LITERAL:featureName KW_WITH opt_properties:properties
+    {:
+        RESULT = new EnableFeatureClause(featureName, properties);
```
Comment (yangzhg) [inline fe/fe-core/src/main/cup/sql_parser.cup:986]: you may add an opt_enable_feture_properties and merge them  to `KW_ENABLE KW_FEATURE STRING_LITERAL:featureName opt_enable_feture_properties`

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStmt.java [hunk 3]
```diff
@@ -79,11 +82,28 @@ public void analyze(Analyzer analyzer) throws UserException {
...
-            if (alterClause instanceof EnableFeatureClause
-                    && ((EnableFeatureClause) alterClause).getFeature() == EnableFeatureClause.Features.BATCH_DELETE) {
+            if (alterClause instanceof EnableFeatureClause) {
+                EnableFeatureClause.Features alterFeature  = ((EnableFeatureClause) alterClause).getFeature();
+                if (alterFeature == null || alterFeature == EnableFeatureClause.Features.UNKNOWN) {
+                    throw new AnalysisException("unknown feature for alter clause");
+                }
                 if (table.getKeysType() != KeysType.UNIQUE_KEYS) {
-                    throw new AnalysisException("Batch delete only supported in unique tables.");
+                    throw new AnalysisException("enable feature only supported in unique tables now.");
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStmt.java:91]: batch delete only  support UNIQUE_KEYS but other may support all tables

File: fe/fe-core/src/main/java/org/apache/doris/analysis/EnableFeatureClause.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/EnableFeatureClause.java]: Done

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 2 above)
Comment (Youngwb) [reply fe/fe-core/src/main/cup/sql_parser.cup]: Done

**Review by yangzhg — DISMISSED** [review]
+1

**Comment by morningman** [comment]
hi @Youngwb , could you please add document for this new feature?

**Comment by Youngwb** [comment]
@morningman  I have updated the document

**Review by yangzhg — APPROVED** [review]
+1

**Pull request merged**